File: UI/src/app/models/request/requestTags.ts
// Tagging request payloads — mirrors UI/src-tauri/src/models/request/request_tags.rs.

export interface requestTagsMutate {
  paths: string[];
  category: string;
  value: string;
}

export interface tagFilterDto {
  category: string;
  value: string;
}

export interface requestTagsQuery {
  filters: tagFilterDto[];
}

File: UI/src-tauri/src/models/request/request_tags.rs
// Tagging request payloads.
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestTagsMutate {
    pub paths: Vec<String>,
    pub category: String,
    pub value: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TagFilterDto {
    pub category: String,
    pub value: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RequestTagsQuery {
    pub filters: Vec<TagFilterDto>,
}
